### PR TITLE
[WIP] Client Changes for Multi Battle 

### PR DESCRIPTION
**Pull request by Spandamn**
Wpould appreciate any help with this, simply PR his branch
Works with he [randbats multi laddder](https://github.com/Spandamn/Pokemon-Showdown/tree/multi7)

### CODE REVIEW

**Comment by Spandamn** [comment]
ffs

File: src/panel-battle.tsx [hunk 1]
```diff
@@ -47,7 +49,7 @@ class BattlesPanel extends PSRoomPanel<BattlesRoom> {
 		return <div><a href={`/${battle.id}`} class="blocklink">
 			{minEloMessage && <small style="float:right">({minEloMessage})</small>}
 			<small>[{format}]</small><br />
-			<em class="p1">{battle.p1}</em> <small class="vs">vs.</small> <em class="p2">{battle.p2}</em>
+			<em class="p1">{battle.p1}</em> & <em class="p3">{battle.p3}</em> <small class="vs">vs.</small> <em class="p2">{battle.p2}</em> & <em class="p4">{battle.p4}</em>
```
Comment (Zarel) [inline src/panel-battle.tsx:52]: This still needs to support non-multi battles.

Comment (Zarel) [reply src/panel-battle.tsx:52]: You're probably better off reverting this part if you don't want to write the specialized code for it.

Comment (Zarel) [reply src/panel-battle.tsx:52]: Or this is a simple fix:

```ts
const p1 = battle.p1 + (battle.p3 ? " & " + battle.p3 : "");
const p2 = battle.p2 + (battle.p4 ? " & " + battle.p4 : "");
```

```suggestion
			<em class="p1">{p1}</em> <small class="vs">vs.</small> <em class="p2">{p2}</em>
```

File: src/battle-text-parser.ts [hunk 2]
```diff
@@ -234,8 +234,10 @@ class BattleTextParser {
 		if (!pokemon) return '';
 		let side;
 		switch (pokemon.slice(0, 2)) {
-		case 'p1': side = 0; break;
-		case 'p2': side = 1; break;
+		case 'p1': case 'p3':
+			side = 0; break;
+		case 'p2': case 'p4':
+			side = 1; break;
```
Comment (Zarel) [inline src/battle-text-parser.ts:240]:
```suggestion
		case 'p1': case 'p3': side = 0; break;
		case 'p2': case 'p4': side = 1; break;
```

**Comment by Zarel** [comment]
@Spandamn you missed this part of my post:

```js
const p1 = battle.p1 + (battle.p3 ? " & " + battle.p3 : "");
const p2 = battle.p2 + (battle.p4 ? " & " + battle.p4 : "");
```

**Comment by Spandamn** [comment]
I'll do that change soon after in done with my lecture

**Comment by Spandamn** [comment]
Hey @Zarel can you committ that im in a bit of troublle

**Comment by Zarel** [comment]
What do you mean by "bit of trouble"?

**Comment by Zarel** [comment]
Oh no! Sorry to hear that. Sure, I can take care of this for you.

**Comment by Spandamn** [comment]
Thanks for understanding. its a difficult time, but if dont keep myself productive, i will fall.
Im going to check this update out, is anyone interested in contributing? I really feel multibattle before 2021 would bbe great,
Random Multi Battles already work.

**Comment by Zarel** [comment]
Let's accept this for now (so people can test on the official client). Any further changes can be a new PR.

**Pull request merged**